src/Event.cs

```diff
@@ -62,41 +62,118 @@ public SubscriptionBuilder<EventContext> OnError(Callback callback)
             return this;
         }
 
-        public SubscriptionHandle<EventContext> Subscribe(params string[] querySqls) => new(conn, Applied, Error, querySqls);
+        public SubscriptionHandle<EventContext> Subscribe(string querySql) => new(conn, Applied, Error, querySql);
 
         public void SubscribeToAllTables()
         {
-            Subscribe("SELECT * FROM *");
+            // make sure we use the legacy handle constructor here, even though there's only 1 query.
+            new SubscriptionHandle<EventContext>(conn, Applied, Error, new string[] { "SELECT * FROM *" });
         }
     }
 
     public interface ISubscriptionHandle
     {
         void OnApplied(IEventContext ctx);
+        void OnError(IEventContext ctx);
+        void OnEnded(IEventContext ctx);
     }
 
     public class SubscriptionHandle<EventContext> : ISubscriptionHandle
         where EventContext : IEventContext
     {
         private readonly SubscriptionBuilder<EventContext>.Callback? onApplied;
+        private readonly SubscriptionBuilder<EventContext>.Callback? onError;
+        private SubscriptionBuilder<EventContext>.Callback? onEnded;
+
+        public bool IsEnded
+        {
+            get; private set;
+        }
+
+        public bool IsActive
+        {
+            get; private set;
+        }
```

Comment:
Nit: I wonder if these should be backed by a private enum state like `{ Initial, Active, Ended }` to avoid accidental confusion in `IsActive` / `IsEnded` assignments.

Reply:
I was originally doing fancier logic here to deal with disconnect being called multiple times, then I gave up on that and just throw an error, this is a remnant of that. Will simplify.